### PR TITLE
Avoid warnings about PATH with pipx installations

### PR DESCRIPTION
**Pull request by ssbarnea**
Fixes: #3919

### CODE REVIEW

**Review by ajinkyau — APPROVED** [review]

File: src/ansiblelint/__main__.py [hunk 1]
```diff
@@ -452,6 +452,7 @@ def path_inject(own_location: str = "") -> None:
         inject_paths.append(str(py_path))
 
     # last option, if nothing else is found, just look next to ourselves...
+    own_location = os.path.realpath(own_location)
```
Comment (FliegenKLATSCH) [inline src/ansiblelint/__main__.py:455]: This breaks #3903 again, do we have to resolve symlinks here? Maybe the resolved location can only be used where needed?

**Comment by yshollander-selerity** [comment]
I think I'm having the same issue on MacOS, ansible-lint installed from Homebrew. ansible-lint is in my PATH, so there should be no warning here.

```
ansible % ansible-lint -vv myplaybookyml 2>&1 | egrep -v 'DEBUG|INFO'
WARNING: PATH altered to include /opt/homebrew/Cellar/ansible-lint/24.2.2/libexec/bin, /opt/homebrew/Cellar/ansible-lint/24.2.2/libexec/bin :: This is usually a sign of broken local setup, which can cause unexpected behaviors.
ansible % which ansible-lint
/opt/homebrew/bin/ansible-lint
```

**Comment by andrewkendall** [comment]
> I think I'm having the same issue on MacOS, ansible-lint installed from Homebrew. ansible-lint is in my PATH, so there should be no warning here.
> 
> ```
> ansible % ansible-lint -vv myplaybookyml 2>&1 | egrep -v 'DEBUG|INFO'
> WARNING: PATH altered to include /opt/homebrew/Cellar/ansible-lint/24.2.2/libexec/bin, /opt/homebrew/Cellar/ansible-lint/24.2.2/libexec/bin :: This is usually a sign of broken local setup, which can cause unexpected behaviors.
> ansible % which ansible-lint
> /opt/homebrew/bin/ansible-lint
> ```

Same here, just updated to 24.2.2 via brew and getting the same error.

**Comment by andrewkendall** [comment]
@yshollander-selerity for a quick fix I've rolled back to an old version. Download 

`https://raw.githubusercontent.com/Homebrew/homebrew-core/9238b09cb713f6ccb13c3d1f1c67a61a04ab4db5/Formula/a/ansible-lint.rb`

cd the directory that is in and run:

`brew install ansible-lint.rb`

**Pull request merged**